### DISCUSSION TITLE
PostgresException: 53300: sorry, too many clients

### DISCUSSION THREAD

**Discussion by ulaskalok** [JasperFx/marten#1493] [Q&A] [answered]
**I didn't do anything special. But I get this error after a few queries.**
` at Npgsql.NpgsqlConnector.<>c__DisplayClass161_0.<<ReadMessage>g__ReadMessageLong|0>d.MoveNext() in C:\\projects\\npgsql\\src\\Npgsql\\NpgsqlConnector.cs:line 1003\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at Npgsql.NpgsqlConnector.<>c__DisplayClass161_0.<<ReadMessage>g__ReadMessageLong|0>d.MoveNext() in C:\\projects\\npgsql\\src\\Npgsql\\NpgsqlConnector.cs:line 1032\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at Npgsql.NpgsqlConnector.Authenticate(String username, NpgsqlTimeout timeout, Boolean async) in C:\\projects\\npgsql\\src\\Npgsql\\NpgsqlConnector.Auth.cs:line 26\r\n   at Npgsql.NpgsqlConnector.Open(NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken) in C:\\projects\\npgsql\\src\\Npgsql\\NpgsqlConnector.cs:line 425\r\n   at Npgsql.ConnectorPool.AllocateLong(NpgsqlConnection conn, NpgsqlTimeout timeout, Boolean async, CancellationToken cancellationToken) in C:\\projects\\npgsql\\src\\Npgsql\\ConnectorPool.cs:line 246\r\n   at Npgsql.NpgsqlConnection.<>c__DisplayClass32_0.<<Open>g__OpenLong|0>d.MoveNext() in C:\\projects\\npgsql\\src\\Npgsql\\NpgsqlConnection.cs:line 301\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at Marten.Services.ManagedConnection.<>c__DisplayClass14_0.<<buildConnectionAsync>b__0>d.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at Marten.Services.ManagedConnection.buildConnectionAsync(CancellationToken token)\r\n   at Marten.Services.ManagedConnection.ExecuteAsync[T](NpgsqlCommand cmd, Func`3 func, CancellationToken token)\r\n   at Marten.Services.CommandRunnerExtensions.FetchAsync[T](IManagedConnection runner, IQueryHandler`1 handler, IIdentityMap map, QueryStatistics stats, ITenant tenant, CancellationToken token)\r\n `

**Comment by ulaskalok** [comment]
I use a genericrepository Each query opens a new session in postgresql.So I get this error after a while.

**Comment by jeremydmiller** [comment]
@ulaskalok Are you disposing the opened sessions?

**Comment by jokokko** [comment, marked as answer]
@ulaskalok "after a few queries" is certainly not normal, unless otherwise odd PG setup or overall heavy traffic. As suggested above, see how you're managing your connections. If needed, take a mem dump and inspect `NpgsqlConnection` instances.

If needed, make use of `pg_stat_activity` or ask your DBA to do so (https://www.postgresql.org/docs/current/monitoring-stats.html#PG-STAT-ACTIVITY-VIEW). From thereon, if needed, proceed to harness https://www.pgbouncer.org/ if you haven't and see it appropriate.
